tools/frr

```diff
@@ -200,40 +200,59 @@ stop()
     fi
 
     if ! started "$inst" ; then
-        echo -n " ($inst)"
+        echo -n "($inst) "
         return 0
+
+    # Make sure that watch FRR is stopped first
+    elif [ $inst == "watchfrr" ]; then
+        PIDFILE=`pidfile $inst`
+        PID=`cat $PIDFILE 2>/dev/null`
+
+	# Send signal to die
+        kill -2 $PID 2>/dev/null &
+
+        # Ask waiter to... wait
+	stop_waiter $PID $inst
     else
         PIDFILE=`pidfile $inst`
         PID=`cat $PIDFILE 2>/dev/null`
-        kill -2 $PID 2>/dev/null
-        #
-        #       Now we have to wait until $DAEMON has _really_ stopped.
-        #
-        if test -n "$PID" && kill -0 $PID 2>/dev/null; then
-            echo -n " (waiting) ."
-            cnt=0
-            while kill -0 $PID 2>/dev/null; do
-                cnt=`expr $cnt + 1`
-                if [ $cnt -gt 60 ]; then
-                    # Waited 120 secs now, fail.
-                    echo -n "Failed.. "
-                    break
-                fi
-                sleep 2
-                echo -n "."
-                done
-            fi
-        echo -n " $inst"
-        rm -f `pidfile $inst`
-        rm -f `vtyfile $inst`
 
-	if [ "$1" = "zebra" ]; then
-	    echo -n "Stopping staticd since zebra is running"
-	    stop staticd
-	fi
+	# Send signal to die
+        kill -2 $PID 2>/dev/null &
+
+        # Ask waiter to... wait
+	stop_waiter $PID $inst &
     fi
 }
 
+stop_waiter(){
+	PID=$1
+	inst=$2
+	if [ "$2" = "zebra" ]; then
+            echo -n "$inst. "
+	            sleep 0.5
+	    echo
+	    echo "Stopping staticd since zebra is running. "
+	fi
+	if [ "$2" != "zebra" ]; then
```

Comment:
Why would we want to treat zebra differently here now?

Reply:
This was done to make it appear last - when run all at once this job was least consistent in how it would present and often wound up with ugliness.